render/json/default.py

```diff
@@ -223,6 +223,27 @@ def renderEntry(self, skel, actionName, params = None):
 		request.current.get().response.headers["Content-Type"] = "application/json"
 		return json.dumps(res)
 
+	def preprocessParams(self, params):
+		"""
+		Translate params to support multilingual categories and tooltips.
+
+		:param params: Params dictionary which values should be translated. If we get no dictionary, we do nothing.
+		:type params: dict
+
+		:return: Params dictionary with translated values.
+		:rtype: dict
+		"""
+
+		if not isinstance(params, dict):
+			return params
+
+		res = params.copy()
```

Comment:
This is ok, but the whole function can also be solved in one-liner
```python
params = {key: _(value) for key, value in params.items() if key in ["category", "tooltip"]}
```
with the copy-stuff, it would be a three liner ;-)

Reply:
This one-line will only create a dictionary with the keys category/tooltip. Others will be ignored.

Reply:
yes, I said it's incomplete.
Here's the full code:

```python
def preprocessParams(self, params):
		"""
		Translate params to support multilingual categories and tooltips.
		:param params: Params dictionary which values should be translated. If we get no dictionary, we do nothing.
		:type params: dict
		:return: Params dictionary with translated values.
		:rtype: dict
		"""

		if not isinstance(params, dict):
			return params

		params = params.copy()
		params.update({key: _(value) for key, value in params.items() if key in ["category", "tooltip"]}

		return params
```

Reply:
`return {key: (_(value) if key in ["category", "tooltip"] else value) for key, value in params.items() }` ? :smile: 

No seriously, this can break a lot of stuff. None of this had been ever translated (also not from jinja2 render) and i've seen a lot of code in projects that depends on the category being stable. This is probably a bigger issue we've have to address in Py3.

Reply:
> `return {key: (_(value) if key in ["category", "tooltip"] else value) for key, value in params.items() }` ? smile

Perfect!